statistics_service/native/src/impl/StatisticsService.hpp

```diff
@@ -1,63 +1,80 @@
+#ifndef STATISTICS_SERVICE_H_
+#define STATISTICS_SERVICE_H_
 
-#ifndef STATISTICS_SERVICE
-#define STATISTICS_SERVICE
 
-#include <aidl/statistics/service/api/BnStatisticsService.h>
 #include <vector>
 
+#include <aidl/statistics/service/api/BnStatisticsService.h>
 
 namespace statistics {
-    namespace service {
-
-        class StatisticsService
-                : public ::aidl::statistics::service::api::BnStatisticsService
-        {
-        public:
-            StatisticsService();
-            ~StatisticsService();
-
-            static ::std::shared_ptr<StatisticsService> getInstance();
-
-
-            // IExampleService AIDL interface callbacks
-            ::ndk::ScopedAStatus getCpuTemperature(float* value_return) override;
-            ::ndk::ScopedAStatus getGpuTemperature(float* value_return) override;
-            ::ndk::ScopedAStatus getAmbientTemperature(float* value_return) override;
-            ::ndk::ScopedAStatus getAverageCpuTemperature(float* value_return) override;
-            ::ndk::ScopedAStatus getAverageGpuTemperature(float* value_return) override;
-            ::ndk::ScopedAStatus getAverageAmbientTemperature(float* value_return) override;
-            ::ndk::ScopedAStatus getMaxCpuTemperature(float* value_return) override;
-            ::ndk::ScopedAStatus getMaxGpuTemperature(float* value_return) override;
-            ::ndk::ScopedAStatus getMaxAmbientTemperature(float* value_return) override;
-
-
-            const std::string getServiceName(void)
-            {
-                return std::string() + descriptor + "/default";
-            }
-
-        private:
-            static ::std::shared_ptr<StatisticsService> S_INSTANCE;
-            static int current_line_to_read;
-            const int TOTAL_NR_LINES = 100;
-
-            vector<float> readTemperatures();
-            float getValue(CallbacksId id);
-
-            enum class CallbacksId {
-                CpuTemperature = 1,
-                GpuTemperature,
-                AmbientTemperature,
-                AverageCpuTemperature,
-                AverageGpuTemperature,
-                AverageAmbientTemperature,
-                MaxCpuTemperature,
-                MaxGpuTemperature,
-                MaxAmbientTemperature,
-            };
+namespace service {
+
+using ::aidl::statistics::service::api::BnStatisticsService;

```

Comment:
Este descurajata folosirea de instructiuni `using` in headers, si in namespace-uri de obicei.
Aceste instructiuni "polueaza" namespace-ul cu aceste tipuri, si pot crea probleme dificil de investigat cand exista coliziuni de nume. Recomandarea este sa folosesti fully qualified names in headers, si using-urile din cpp sa nu fie parte din namespace.

https://google.github.io/styleguide/cppguide.html#Namespaces
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#sf6-use-using-namespace-directives-for-transition-for-foundation-libraries-such-as-std-or-within-a-local-scope-only

Reply:
si in cpp as putea sa pun acele using-uri intr-un namespace anonim imediat in interiorul ultimului namesapce sau ar trebui sa le pun in afara setului de namesapce-uri?

Reply:
Este recomandat sa folosesti using directives, doar in fisierele C++. fie in global space sau in anonymous namespace(cred ca e basically echivalent).
Anoynmous namespace este un namespace fara nume definit in global scope(nu in alt namespace).